File: VB/GridColumnHeight.Module.Win/GridHeaderHeightController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.Persistent.Base
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports XAFCommunity.Win

Namespace GridColumnHeight.Module.Win
	Partial Public Class GridHeaderHeightController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)

			TargetViewType = ViewType.ListView
		End Sub

		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			AddHandler View.ControlsCreated, AddressOf View_ControlsCreated
		End Sub

		Protected Overrides Sub OnDeactivated()
			RemoveHandler View.ControlsCreated, AddressOf View_ControlsCreated
			MyBase.OnDeactivated()
		End Sub

		Private Sub View_ControlsCreated(ByVal sender As Object, ByVal e As EventArgs)
            Dim modelListViewHeaderRows As IModelListViewHeaderRows = CType(Me.View.Model, IModelListViewHeaderRows)
            If modelListViewHeaderRows IsNot Nothing Then
                SetColumnHeadingRowHeight(TryCast(View, ListView), modelListViewHeaderRows.HeaderRows)
            End If
		End Sub

		''' <summary>
		''' Configures the supplied ListView grid to show word wrapped column captions.
		''' </summary>
		''' <param name="view"></param>
		''' <param name="lineCount">The number of lines of text required in the column header.</param>
		Friend Shared Sub SetColumnHeadingRowHeight(ByVal view As ListView, ByVal lineCount As Integer)
			If view IsNot Nothing AndAlso lineCount > 1 Then
				Dim listEditor As GridListEditor = TryCast(view.Editor, GridListEditor)
				If listEditor IsNot Nothing Then
					GridHeaderHeightHelper.SetGridEditorHeaderHeight(listEditor, lineCount)
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/GridColumnHeight.Module.Win/GridHeaderHeightController.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.Persistent.Base;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using XAFCommunity.Win;

namespace GridColumnHeight.Module.Win
{
    public partial class GridHeaderHeightController : ViewController
    {
        public GridHeaderHeightController()
        {
            InitializeComponent();
            RegisterActions(components);

            TargetViewType = ViewType.ListView;
        }

        protected override void OnActivated()
        {
            base.OnActivated();
            View.ControlsCreated += View_ControlsCreated;
        }

        protected override void OnDeactivating()
        {
            View.ControlsCreated -= View_ControlsCreated;
            base.OnDeactivating();
        }

        void View_ControlsCreated(object sender, EventArgs e)
        {
            DictionaryAttribute attr = this.View.Info.GetAttribute("HeaderRows");
            if (attr!= null) 
            {
                int lines;
                if (int.TryParse(attr.Value, out lines))
                {
                    SetColumnHeadingRowHeight(View as ListView, lines);
                }
            }
        }

        /// <summary>
        /// Configures the supplied ListView grid to show word wrapped column captions.
        /// </summary>
        /// <param name="view"></param>
        /// <param name="lineCount">The number of lines of text required in the column header.</param>
        internal static void SetColumnHeadingRowHeight(ListView view, int lineCount)
        {
            if (view != null && lineCount > 1)
            {
                GridListEditor listEditor = view.Editor as GridListEditor;
                if (listEditor != null)
                    GridHeaderHeightHelper.SetGridEditorHeaderHeight(listEditor, lineCount);
            }
        }
    }
}